packages/core/src/config/utilityTypes.ts

```diff
@@ -0,0 +1,51 @@
+import type { Abi, AbiEvent, FormatAbiItem } from "abitype";
+import type { ParseAbiItem } from "viem";
+
+export type NonStrictPick<T, K> = {
+  [P in Extract<keyof T, K>]: T[P];
+};
+
+export type ExtractAbiEvents<
+  abi extends Abi,
+  events = Extract<abi[number], { type: "event" }>,
+> = [events] extends [never] ? AbiEvent : events;
+
+export type ParseAbiEvent<
+  abi extends Abi,
+  signature extends string,
+  ///
```

Comment:
?

Reply:
This is a convention I took from abitype/viem. Basically, the type arguments under the "///" are actually just variables. These arguments should never be passed in and instead the type falls back to the default value assigned by the "=".